Add initial cookbooks and fill-in additional details on processors (#4)
marklogic/nifi: _pages/cookbooks/cookbook-recipes.md

---
layout: inner
title: Cookbook Recipes
lead_text: ''
permalink: /cookbook-recipes
---

# Cookbook Recipes

## [Relational Database][relational-database]

> Pull data into MarkLogic from a relational database source as JSON documents.

Reviewer: Phrasing is a little awkward. Rephrase:

```suggestion
Pull data from a relational database source and write into MarkLogic as JSON documents.
```

[Read More][relational-database]

## [File System][file-system]

Pull data into MarkLogic from a file system directory.

[Read More][file-system]

## [Data Hub Input Flow][dhf-input-flow]

> Use the `Server Transform` property and dynamic `trans:*` properties to run a Data Hub Input Flow transform against the input documents.

Reviewer: Skip the implementation details at this level. 

```suggestion
Use a [MarkLogic Data Hub](https://github.com/marklogic/marklogic-data-hub) Input Flow to transform documents while loading.
```
